packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -82,54 +86,120 @@ import {
   QueryBuilderNavigationBlocker,
   type QueryBuilderState,
 } from '@finos/legend-query-builder';
+import { LegendQueryTelemetryHelper } from '../application/LegendQueryTelemetryHelper.js';
+import type { HelpMenuItemEditorRender } from '../stores/LegendQueryApplicationPlugin.js';
 
 const QueryExportDialogContent = observer(
   (props: { exportState: QueryExportState }) => {
     const { exportState } = props;
     const applicationStore = useApplicationStore();
-    const allowCreate = exportState.allowPersist;
+
+    const [isExistingQueryName, setIsExistingQueryName] = useState(false);
+
+    const saveNewQuery = exportState.allowPersist;
     const allowSave = exportState.allowPersist && exportState.allowUpdate;
     const create = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(true),
     );
-    const save = applicationStore.guardUnhandledError(() =>
+    const updateQuery = applicationStore.guardUnhandledError(() =>
       exportState.persistQuery(false),
     );
+    const renameExport = Boolean(exportState.renameExport);
 
     // name
     const nameInputRef = useRef<HTMLInputElement>(null);
-    const changeName: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+
+    const debouncedLoadQueries = useMemo(
+      () =>
+        debounce((input: string): void => {
+          flowResult(
+            exportState.editorStore.queryLoaderState.loadQueries(input),
+          ).catch(applicationStore.alertUnhandledError);
+        }, 500),
+      [applicationStore, exportState.editorStore.queryLoaderState],
+    );
+
+    const changeName: React.ChangeEventHandler<HTMLInputElement> = (event) => {
       exportState.setQueryName(event.target.value);
+    };
 
     useEffect(() => {
       nameInputRef.current?.focus();
     }, []);
 
+    useEffect(() => {
+      const searchText = exportState.queryName.replace('Copy of ', '');
+      debouncedLoadQueries.cancel();
+      debouncedLoadQueries(searchText);
+      const existingQuery =
+        exportState.editorStore.queryLoaderState.queries.some(
+          (q) => q.name === searchText,
+        );
+
+      if (exportState.queryName.includes('Copy of Copy of') && existingQuery) {
```

Comment:
what is this? I don't think we should handle this

Reply:
I thought we had decided we should only show a caution sign if the user had 'copy of copy of...[og query]' since otherwise they would see yellow caution icon every time they pressed save query?